lib/charms/opensearch/v0/opensearch_tls.py

```diff
@@ -430,7 +427,7 @@ def _create_keystore_pwd_if_not_exists(self, scope: Scope, cert_type: CertType,
         if secrets:
             store_pwd = secrets.get(f"{store_type}-password")
 
-        if not store_pwd:
+        if not store_pwd and not self.charm.opensearch_peer_cm.is_consumer(of="main"):
```

Comment:
Reason for this: on `CertificateAvailable`, the keystore password could be created if it does not exist yet. If this happens before the relation data is updated into the secret, the secret does not get populated anymore with data from the main orchestrator. I saw this fail in various test runs, and there is actually a test failure every now and then in our main branch, where there seems to be a wrong keystore password. I did not understand why this could happen for quite some time, but tracked this down to this now.

Reply:
> the above condition can be simplified to self.charm.opensearch_peer_cm.is_provider(typ="main")

Is this really the same like `not self.charm.opensearch_peer_cm.is_consumer(of="main")`, also in cases where there is no peer-cluster-relation?

Reply:
Added a check for `CertType.APP_ADMIN` to make sure this is done for the `unit-http` and `unit-transport` cases on the consumer-units of large deployments.